certificate/constraints/constraints.go

```diff
@@ -109,3 +108,25 @@ func IsTechnicallyConstrained(cert *x509.Certificate) bool {
 	}
 	return false
 }
+
+// IsTechnicallyConstrainedMozPolicyV2_5 determines if a given certificate is technically constrained
+// according to the Mozilla Root Store Policy V2.5.
+// https://www.mozilla.org/en-US/about/governance/policies/security-group/certs/policy/
+func IsTechnicallyConstrainedMozPolicyV2_5(cert *x509.Certificate) bool {
+	// The logic from IsTechnicallyConstrained is extended here due to paragraph
+	// three of section 5.3.1:
+	//
+	// If the certificate includes the id-kp-emailProtection extended key usage,
+	// it MUST include the Name Constraints X.509v3 extension with constraints on
+	// rfc822Name, with at least one name in permittedSubtrees, each such name having
+	// its ownership validated according to section 3.2.2.4 of the Baseline Requirements.
+	for _, extKeyUsage := range cert.ExtKeyUsage {
+		if extKeyUsage == x509.ExtKeyUsageEmailProtection {
+			if len(cert.PermittedEmailAddresses) == 0 {
+				return false
```

Comment:
Someone else should review this. @jcjones maybe?

Reply:
Indeed. Also, Wayne Thayer gave this an out-of-band `LGTM`.

Reply:
good enough for me then.